### DISCUSSION TITLE
Dangerously close to reinventing the wheel

### DISCUSSION THREAD

**Discussion by gcarreno** [pasichDev/noso-faucet#26] [General]
Hey Andrii (@pasichDev),

Might as well say this now, in these early stages, than say it later when you've invested too much time in this architecture.

With a router(/dispatcher), a semblance of controllers and a very rudimentary layout/templating system, we are now in the middle of `SlimPHP`'s territory.

So my arguments are these:
- Granted, you don't need a framework for this, but by using one you don't have to create and test the boilerplate code for the basics.
- At the moment, your code is functional but lacks any ways of testing it via automation, the Continuous Integration of the CI/CD pair.

Yes, you can curse me and throw a fit of rage, cuz in essence this would be another re-write :frowning_face:
Hence me creating this issue now, so we can discuss this possible change of path.
**Hint**: This should be a discussion and I think you should open that area on the Project settings :smiley: 

Have you ever done any work using the `SlimPHP` framework?
At the moment, `SlimPHP` is the most light weight of all the PHP frameworks, usually used for very simple projects or for rapid prototyping of a possible more complex one.

What are your thoughts on this subject?

Cheers,
Gus
<hr>

1. https://www.slimframework.com/
2. https://github.com/slimphp

**Comment by pasichDev** [comment]
Hi @gcarreno ,
I understand what you mean now. Yes, it is very similar to SlimPHP.
But there is a problem, I have never worked with this framework, I can learn it very quickly, there are no problems with this.
But are the NOSO developers ready to wait for me?

Is using Slim PHP a good idea if you really need it?

**Comment by pasichDev** [comment]
HI @gcarreno , switching to Slim PHP, I already started in a new branch

**Comment by gcarreno** [comment]
Hey Andrii (@pasichDev),

Looks like you got motivated :grinning: 

I'm glad that you finally entered the framework mentality :smiley: 

If you need any help with anything else, please don't hesitate to ask!!

Cheers,
Gus
